docs/how-tos/use-multi-format-reader.md

```diff
@@ -0,0 +1,396 @@
+2) `".yml"`, `".yaml"`: This will be used to parse in the (meta)data from the ELN file.
+3) `".json"`: This will be used to read in the **config file**, which is used to map from the (meta)data concepts from the instrument and ELN data to the concepts in the NXDL file.
+
+## Reading in the instrument's data and metadata
+First, we will have a look at the HDF5 file.
+
+<img src="media/mock_data.png" style="width: 50vw; min-width: 330px;" />
+
+Here, we see that we have a `data` group with x and y values, as well as some additional metadata for the instrument.
+
+Here is one way to implement the method to read in the data:
+```python
+import h5py
+
+def handle_hdf5_file(filepath):
+    def recursively_read_group(group, path=""):
+        result = {}
+        for key, item in group.items():
+            new_path = f"{path}/{key}" if path else key
+            if isinstance(item, h5py.Group):
+                # Recursively read subgroups
+                result.update(recursively_read_group(item, new_path))
+            elif isinstance(item, h5py.Dataset):
+                # Read datasets
+                result[new_path] = item[()]
+        return result
+
+    # Open the HDF5 file and read its contents
+    with h5py.File(filepath, "r") as hdf:
+        self.hdf5_data = recursively_read_group(hdf)
+
+    return {}
+```
+Note that here we are returning an empty dictionary because we don't want to fill the template just yet, but only read in the HDF5 data for now. We will use the config file later to fill the template with the read-in data.
+
+`self.hdf5_data` will look like this:
+```python
+{
+    "data/x_values": array([-10.        ,  -9.7979798 ,  -9.5959596 , ...,  10.        ]),
+    "data/y_values": array([3.72665317e-06, 6.14389891e-06, 1.00262383e-05, ..., 3.72665317e-06]),
+    "data/x_units": "eV",
+    "data/y_units": "counts_per_second",
+    "metadata/instrument/version": 1.0,
+    "metadata/instrument/detector/name": "my_gaussian_detector",
+    "metadata/instrument/detector/count_time": 1.2,
+    "metadata/instrument/detector/count_time_units": s",
+}
+```
+## Reading in ELN data
+As we can see in the application definition `NXsimple` above, there are some concepts defined for which there is no equivalent metadata in the HDF5 file. We are therefore using a YAML ELN file to add additional metadata.
+The ELN file `eln_data.yaml` looks like this:
+```yaml
+title: My experiment
+user:
+  name: John Doe
+  address: 123 Science Rd, Data City, DC
+sample:
+  name: my_sample
+  physical_form: powder
+  temperature:
+    value: 300
+    unit: K
+```
+
+It contains metadata about the user and the sample that was measured.
+
+We now need to write a function to read in this ELN data. Luckily, there exists already a solution within `pynxtools`, using the `parse_yaml` function:
+
+```python
+from pynxtools.dataconverter.readers.utils import parse_yml
+
+CONVERT_DICT = {

```

Comment:
@lukaspie indeed many similarities with concept mapping code in pynxtools-em and pynxtools-apm the tricky parts are cases like
detectorID[detector*], i.e. how to deal with multiple instances of things. I agree that this might be to much to cover for a getting started

Reply:
This is exactly what we were discussing in #250 and I even suggested the same notation. We opted to not go for this for now because the PR was already large. I see two options:
- writing `detectorID[detector*]`, thus saying that `*` is in the key, it has to be overwritten
- simply writing `ENTRY/INSTRUMENT/detectorID`, i.e., if there is no default name, simply not write it at all. Then the reader must implement the replacements (but we could use default `entry/instrument/detector_id` for example).

Reply:
I think we are talking about two different things here:
1) (not) using the `CONVERT_DICT`. I agree that your approach of non-replacing is probably better. I went with using the replacement in the example here so that people can see that you can also use the `key` instead of the `path` passed to `get_eln_data` to fill the template.
2) what @mkuehbach was talking about is the option of having multiple instances of e.g. a detector, which only get detected when handling the file. In that case, it might be a good idea to have a wildcard syntax in the config _key_ that tells the reader that these names must be replaced (this is what I outlined above)